backend/django/core/templates/projects/create/create_wizard_labels.html

```diff
@@ -73,15 +85,60 @@ <h3>Instructions</h3>
 {% endblock %}
 
 {% block scripts_body %}
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-csv/0.71/jquery.csv-0.71.min.js"></script>
 <script type="text/javascript">
-$('.formset_row_label').formset({
-  addText: 'add label',
-  deleteText: 'remove label',
-  prefix: 'label_set',
-  added: function() {
-    // Reset drop down and errors from both columns when adding new rows
-    $('tr.formset_row_label').last().children(':first').find('ul.errorlist li').empty();
+  function handleUpload(event) {
+    if (event.target.files[0] && event.target.files[0].name.includes('.csv')) {
+      $('.overlay').show();
+      $('#loading_text').html("Please wait. Generating labels...");
+      $("#second_progress").addClass("loader");
+      var reader = new FileReader();
+      try {
+        reader.readAsText(event.target.files[0]);
+        reader.onload = function(event){
+          var csv = event.target.result;
+          var data = $.csv.toArrays(csv);
+            data.forEach((label, index) => {
+              if(index !== 0) {
+                if($('.formset_row_label').eq(index - 1).length > 0) {
+                  $('.formset_row_label').eq(index - 1).css('display', '');
+                  $('.formset_row_label').eq(index - 1).find('input[type=text]').eq(0).val(label[0]);
+                  if(label[1].length > 0) {
+                    $('.formset_row_label').eq(index - 1).find('textarea').eq(0).val(label[1]);
+                  } else {
+                    $('.formset_row_label').eq(index - 1).find('textarea').eq(0).val('');
+                  }
+                } else {
+                  $(':contains("add label")').click();
+                  $('.formset_row_label').last().find('input[type=text]').eq(0).val(label[0]);
+                  if(label[1].length > 0) {
+                    $('.formset_row_label').last().find('textarea').eq(0).val(label[1]);
+                  }
+                }
+              }
+            });
+            if($('.formset_row_label').length > data.length -1) {
+              $('.formset_row_label').slice(data.length - 1).remove();
+            }
+            $('.overlay').hide();
+        }
+      } catch (error) {
+        $('.error-messages').empty();
+        $('.error-messages').eq(1).append('<ul class="error-list"><li>There was an error processing your label file. Please try again.</li></ul>');
+      }
+    } else {
+      $('.error-messages').empty();
+      $('.error-messages').eq(1).append('<ul class="error-list"><li>There was an error processing your label file. Please make sure your file is a .csv and has a Label and Description column.</li></ul>');
```

Comment:
So since we have it available for the data upload page, we should probably have a sample label file for people to download from the label upload page.

Also just noticed that the sample data file for the data page is using the old data source...-_-,

Reply:
I went ahead and made sample label .csv and updated the example data to have the new r/awww data